### PR TITLE
KDS-1865-fix-design-tokens-dep-in-kaio

### PR DESCRIPTION
**Pull request by gyfchong**
## Why
Hard to explain but having `design-tokens` as a peer-dep causes unnecessary complexities.

Additionally, we have no need for the Theme function anymore.

## What
Remove dependency on `design-tokens` by removing any theme related functionality, only using `design-tokens` as part of our bundling.

### CODE REVIEW

File: packages/components/src/KaizenProvider/KaizenProvider.tsx [hunk 1]
```diff
@@ -1,21 +1,21 @@
 import React from "react"
-import { Theme as BaseTheme } from "@kaizen/design-tokens"
-import { OptionalIntlProvider } from "./OptionalIntlProvider"
-import { ThemeProvider } from "./ThemeProvider"
+import { FontDefinitions } from "./subcomponents/FontDefinitions"
+import { OptionalIntlProvider } from "./subcomponents/OptionalIntlProvider"
 
-export type KaizenProviderProps<Theme extends BaseTheme = BaseTheme> = {
+export type KaizenProviderProps = {
```
Comment (HeartSquared) [inline packages/components/src/KaizenProvider/KaizenProvider.tsx:5]: So, breaking change? Is it just fixing the next-template?

File: packages/components/src/KaizenProvider/_docs/KaizenProvider.mdx [hunk 2]
```diff
@@ -25,13 +25,10 @@ import { KaizenProvider } from "@kaizen/components";
...
-### ThemeManager support
-- [Managing the theme](/story/components-kaizen-provider-managing-the-theme--docs)
-
 ### Internationalization support
 - [Internationalization in Kaizen](/story/components-kaizen-provider-internationalization-in-kaizen--docs)
 
-## Migration from ThemeProvider to KaizenProvider v1.0
+## Migrating to KaizenProvider v1.0
 
 `KaizenProvider` at this version is a direct replacement for `ThemeProvider`, and is safe to do a 1 to 1 replacement.
```
Comment (HeartSquared) [inline packages/components/src/KaizenProvider/_docs/KaizenProvider.mdx:33]: This needs updating.

File: packages/components/src/KaizenProvider/_docs/managing-the-theme.mdx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Meta } from "@storybook/blocks"
 
 # Managing the theme
 
-Out of the box, the `KaizenProvider` utilises the `defaultTheme` from `@kaizen/design-tokens`. However if you want to roll your own theme, we provide a `ThemeManager` function which can apply your custom theme.
+Out of the box, the `KaizenProvider` sets the `defaultTheme`. However if you want to roll your own theme, you can construct a Theme object and pass it in.
```
Comment (HeartSquared) [inline packages/components/src/KaizenProvider/_docs/managing-the-theme.mdx]: This needs updating to match the other docs.

File: packages/components/src/utils/useMediaQueries.tsx [hunk 4]
```diff
@@ -45,7 +44,8 @@ export const useMediaQueries = (
       },
     }
   }
-  const theme = useTheme()
+
+  const breakpoints = { medium: "768px", large: "1080px" }
```
Comment (HeartSquared) [inline packages/components/src/utils/useMediaQueries.tsx:48]: Given it's meant to match values in our CSS vars, probably worth leaving a comment here and in the tokens source (if possible)

File: packages/design-tokens/css/variables.css [hunk 5]
```diff
@@ -0,0 +1 @@
...hy-paragraph-small-font-size: 0.875rem; --typography-paragraph-small-line-height: 1.125rem; --typography-paragraph-small-letter-spacing: normal; --typography-paragraph-small-max-width: 680px; --typography-paragraph-extra-small-font-family: "Inter", "Noto Sans", Helvetica, Arial, sans-serif; --typography-paragraph-extra-small-font-weight: 400; --typography-paragraph-extra-small-font-size: 0.75rem; --typography-paragraph-extra-small-line-height: 1.125rem; --typography-paragraph-extra-small-letter-spacing: normal; --typography-paragraph-extra-small-max-width: 600px; --typography-paragraph-bold-font-weight: 600; --typography-button-primary-font-family: "Inter", "Noto Sans", Helvetica, Arial, sans-serif; --typography-button-primary-font-weight: 700; --typography-button-primary-font-size: 1.125rem; --typography-button-primary-line-height: 1.5rem; --typography-button-primary-letter-spacing: normal; --typography-button-secondary-font-family: "Inter", "Noto Sans", Helvetica, Arial, sans-serif; --typography-button-secondary-font-weight: 500; --typography-button-secondary-font-size: 1rem; --typography-button-secondary-line-height: 1.5rem; --typography-button-secondary-letter-spacing: normal;}
```
Comment (HeartSquared) [inline packages/design-tokens/css/variables.css:1]: Whoah. Is this a generated file we can update easily? Or a source file that we maintain (we'd want more line breaks if so)?

File: packages/design-tokens/bin/buildCssVarTokens.ts [hunk 6]
```diff
@@ -29,8 +30,18 @@ const run = async (): Promise<void> => {
...
+  )
+
+  fs.writeFileSync(
+    path.resolve(cssOutput, "variables.css"),
+    `html {${cssVars}}`
+  )
+
   /**
-   *
+   * augmentedThemeWithCSSVariableValuesVersion
```
Comment (HeartSquared) [inline packages/design-tokens/bin/buildCssVarTokens.ts:44]: Oh? Is this referencing a function?

File: packages/design-tokens/docs/pages/how-to-use-in-js.mdx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+
```
Comment (HeartSquared) [inline packages/design-tokens/docs/pages/how-to-use-in-js.mdx:1]:
```suggestion
```

File: packages/design-tokens/docs/pages/how-to-use-in-js.mdx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import { Meta } from "@storybook/blocks"
+import { LinkTo } from "~storybook/components/LinkTo"
+
+<Meta title="Systems/Tokens/Access tokens via JS" parameters={{ docsOnly: true }} />
+
+# Access tokens via JSX
+
+## 1. Set up a KaizenProvider
+
+Ensure you have followed the steps in <LinkTo pageId="components-kaizen-provider-installation">KaizenProvider</LinkTo> to set up your application for using themes.
```
Comment (HeartSquared) [inline packages/design-tokens/docs/pages/how-to-use-in-js.mdx:11]:
```suggestion
Ensure you have followed the steps in <LinkTo pageId="components-kaizen-provider-installation">KaizenProvider</LinkTo> to set up your application for using design tokens.
```

File: packages/design-tokens/tokens/animation.json.d.ts [hunk 9]
```diff
@@ -1,11 +1,5 @@
 import { DeepMapObjectLeafs, Theme } from "../src/types"
-/**
```
Comment (HeartSquared) [inline packages/design-tokens/tokens/animation.json.d.ts:2]: I wonder when we can remove these manually created `.d.ts` files

File: packages/components/package.json [hunk 10]
```diff
@@ -71,7 +70,9 @@
     "@react-stately/select": "^3.5.5",
     "@react-types/shared": "^3.21.0",
     "classnames": "^2.3.2",
+    "color-string": "^1.9.1",
     "date-fns": "^2.30.0",
+    "lodash.kebabcase": "^4.1.1",
```
Comment (HeartSquared) [inline packages/components/package.json:75]: This appears to only be used in `design-tokens`

File: packages/components/package.json [hunk 11]
```diff
@@ -71,7 +70,9 @@
     "@react-stately/select": "^3.5.5",
     "@react-types/shared": "^3.21.0",
     "classnames": "^2.3.2",
+    "color-string": "^1.9.1",
```
Comment (HeartSquared) [inline packages/components/package.json:73]: This appears only used in `design-tokens` and some global docs (therefore goes on the root as a devDep)

File: packages/components/package.json [hunk 12]
```diff
@@ -98,6 +99,7 @@
     "@rollup/plugin-json": "^6.0.1",
     "@rollup/plugin-node-resolve": "^15.2.3",
     "@rollup/plugin-typescript": "^11.1.5",
+    "@types/color-string": "^1.5.4",
```
Comment (HeartSquared) [inline packages/components/package.json:102]: Follows `color-string`

File: packages/components/src/KaizenProvider/KaizenProvider.tsx (hunk 1 above)
Comment (gyfchong) [reply packages/components/src/KaizenProvider/KaizenProvider.tsx:5]: Going to sneak this in, cause no one uses the `theme` prop on `KaizenProvider` 😄

Comment (HeartSquared) [reply packages/components/src/KaizenProvider/KaizenProvider.tsx:5]: 🥷 🙈

File: packages/design-tokens/css/variables.css (hunk 5 above)
Comment (gyfchong) [reply packages/design-tokens/css/variables.css:1]: It's generated via the build command!

File: packages/design-tokens/bin/buildCssVarTokens.ts (hunk 6 above)
Comment (gyfchong) [reply packages/design-tokens/bin/buildCssVarTokens.ts:44]: No idea haha, left it as is.

File: packages/design-tokens/css/variables.css (hunk 5 above)
Comment (HeartSquared) [reply packages/design-tokens/css/variables.css:1]: Would it be easy to add a comment at the top to say it's auto generated? If not don't worry about it :)

File: packages/design-tokens/bin/buildCssVarTokens.ts (hunk 6 above)
Comment (HeartSquared) [reply packages/design-tokens/bin/buildCssVarTokens.ts:44]: Really? Cos it looks like you added it in 😂

Comment (gyfchong) [reply packages/design-tokens/bin/buildCssVarTokens.ts:44]: Oh wait, yeah it is below

File: packages/design-tokens/css/variables.css (hunk 5 above)
Comment (gyfchong) [reply packages/design-tokens/css/variables.css:1]: Ah can do, done.

File: packages/design-tokens/docs/pages/how-to-use-in-js.mdx [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+### In JS functions
+
+You can access the tokens directly:
+
+```jsx
+import { tokens } from "@kaizen/design-tokens/js"
+
+const myAmazingFunction = () => {
+  const mainColor = data ? colors.purple[500]: colors.purple[300];
```
Comment (HeartSquared) [inline packages/design-tokens/docs/pages/how-to-use-in-js.mdx:39]: Should be this I think? Unless the import is the place to be changed.
```suggestion
  const mainColor = data ? tokens.colors.purple[500]: tokens.colors.purple[300];
```

File: packages/design-tokens/src/types.ts [hunk 14]
```diff
@@ -20,6 +20,9 @@ type Hex = string
 // Once TypeScript is upgraded in the repo, you can use Lowercase<>
 // type Hex = Lowercase<string>
 
+/**
+ * @deprecated Not needed if you are using `KaizenProvider` from `@kaizen/components` or `defaultPreset` from next-services.
+ */
 export type TypographyFont = {
```
Comment (HeartSquared) [inline packages/design-tokens/src/types.ts:26]: Most of these are duplicated in `js/types.ts` and exported. TS conflict, so only keep one of each (or don't export in one file).

Comment (gyfchong) [reply packages/design-tokens/src/types.ts:26]: Shouldn't clash, cause they're not being exported from the same index?

Comment (HeartSquared) [reply packages/design-tokens/src/types.ts:26]: Ohh interesting. Yeah sorry was looking at the wrong barrel file and thought they were being exported together.
I guess it's fine? But given the ones that are exported at the root are the deprecated ones, would that cause confusion?

File: packages/components/package.json [hunk 15]
```diff
@@ -98,6 +97,7 @@
...
```
Comment (HeartSquared) [inline packages/components/package.json:100]: This still needs to be moved to root package.json :P

**Review by HeartSquared — DISMISSED** [review]

File: packages/design-tokens/src/types.ts (hunk 14 above)
Comment (gyfchong) [reply packages/design-tokens/src/types.ts:38]: Mmm risk I'm willing to take haha

**Review by HeartSquared — APPROVED** [review]

**Pull request merged**